cli/tests/test_global_opts.rs

```diff
@@ -774,6 +777,105 @@ fn test_conditional_config() {
     ");
 }
 
+/// Test that `jj` command works with the default configuration.
+#[test]
+fn test_default_config() {
+    let mut test_env = TestEnvironment::default();
+    let config_dir = test_env.env_root().join("empty-config");
+    std::fs::create_dir(&config_dir).unwrap();
+    test_env.set_config_path(&config_dir);
+
+    let envs_to_drop = test_env
+        .jj_cmd(test_env.env_root(), &[])
+        .get_envs()
+        .filter_map(|(name, _)| name.to_str())
+        .filter(|&name| name.starts_with("JJ_") && name != "JJ_CONFIG")
+        .map(|name| name.to_owned())
+        .sorted_unstable()
+        .collect_vec();
+    insta::assert_debug_snapshot!(envs_to_drop, @r#"
+    [
+        "JJ_EMAIL",
+        "JJ_OP_HOSTNAME",
+        "JJ_OP_TIMESTAMP",
+        "JJ_OP_USERNAME",
+        "JJ_RANDOMNESS_SEED",
+        "JJ_TIMESTAMP",
+        "JJ_TZ_OFFSET_MINS",
+        "JJ_USER",
+    ]
+    "#);
+    let jj_cmd_ok = |current_dir: &Path, args: &[&str]| {
+        let mut cmd = test_env.jj_cmd(current_dir, args);
+        for name in &envs_to_drop {
+            cmd.env_remove(name);
+        }
+        let assert = cmd.assert().success();
+        let stdout = test_env.normalize_output(&get_stdout_string(&assert));
+        let stderr = test_env.normalize_output(&get_stderr_string(&assert));
+        (stdout, stderr)
+    };
+
+    let mut insta_settings = insta::Settings::clone_current();
+    insta_settings.add_filter(r"\b[a-zA-Z0-9\-._]+@[a-zA-Z0-9\-._]+\b", "<user>@<host>");
+    insta_settings.add_filter(
+        r"\b[0-9]{4}-[0-9]{2}-[0-9]{2} [0-9]{2}:[0-9]{2}:[0-9]{2}\b",
+        "<date-time>",
+    );
+    insta_settings.add_filter(r"\b[k-z]{8,12}\b", "<change-id>");
+    insta_settings.add_filter(r"\b[0-9a-f]{8,12}\b", "<id>");
+    let _guard = insta_settings.bind_to_scope();
+
+    let is_maskable_char = |c: u8| c.is_ascii_alphanumeric() | matches!(c, b'-' | b'.' | b'_');
+    let is_maskable_str = |s: &str| s.as_bytes().iter().all(|&c| is_maskable_char(c));
+    let maskable_op_user = whoami::fallible::hostname().is_ok_and(|s| is_maskable_str(&s))
+        && whoami::fallible::username().is_ok_and(|s| is_maskable_str(&s));
```

Comment:
Couldn't you concatenate `user`, `@`, and `host`, and use the same regexp as in the `insta` setting to check if it is maskable to ensure consistency?

Reply:
It also works. If insta API accepted `Regex` object, I would do that way. I don't have strong preference over these.

Reply:
Updated to use regex, but the pattern is slightly different. `^..$` vs `\b..\b`.